### PR TITLE
fix sync undefined issue

### PR DESCRIPTION
**Pull request by mengyangwang**
Summary
---------

this PR is to fix the issue that ThreatExchange tab is broken when click sync button. The issue is that we have changed one param name in datasets api, but didn't make all corresponding changes in UI

Test Plan
---------

before fix:
199494540_602475757383245_8708732885548711817_n

after fix:
Screen Shot 2021-06-14 at 12 29 26 PM

### CODE REVIEW

File: hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx
Comment (BarrettOlson) [inline hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx:75]: I'm confused by this line, `setLoading(false)` as a parameter for `fetchAllDatasets.` 
When is it called?

Comment (mengyangwang) [reply hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx]: yeah I think that's wrong, it should not be a parameter, I should call ```setLoading(false)``` before calling ```fetchAllDatasets```, I made the change

File: hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx
Comment (BarrettOlson) [inline hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx:75]: hmm, don't we want the opposite? We are starting a request therefore loading is `true` or am I missing something?

Comment (mengyangwang) [reply hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx:75]: the name might be confusing, because it's equals to hidden, so when starting a request I set hidden as false

Comment (BarrettOlson) [reply hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx:75]: then can we call it hidden or switch the conditional? Otherwise the PR looks good to me :)

**Review by BarrettOlson — APPROVED** [review]
Stamping to unblock. Please swap `Loading`'s values or rename it before merge. Thanks!

File: hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx
Comment (mengyangwang) [reply hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx]: Thanks for pointing that out! I am still using the name loading and I've switched the conditional to ```hidden=!loading```, so it looks more logical

File: hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx
Comment (eriksendc) [inline hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx:152]: I took the new version of this file and put it in my local environment and I'm still getting an error because there is a time when datasets is undefined, and so there won't be a .length property. It's the same error as this morning. I think you need `!datasets || datasets.length === 0` as the condition at line 155. But that then leads to another error at line 177, because again, there is a time when datasets is undefined, so `some` cannot be referenced.

I think you need to do a terraform desctroy then terraform apply to recreate the scenario I'm in.

Comment (BarrettOlson) [reply hasher-matcher-actioner/webapp/src/pages/settings/ThreatExchangeSettingsTab.jsx:152]: `datasets` is initialized to `[]` so I didn't think that would be an issue. Does a bad API request make it undefined? It feels like this check might just hide an API level issues.

**Pull request merged**
